packages/@react-spectrum/overlays/src/Popover.tsx

```diff
@@ -115,10 +115,14 @@ const PopoverWrapper = forwardRef((props: PopoverWrapperProps, ref: RefObject<HT
   }, state);
   let {focusWithinProps} = useFocusWithin(props);
 
+  let onPointerDown = () => {
+    state.close();
+  };
```

Comment:
Moved this change so it is Spectrum specific for our submenu case. Having it as a part of useOverlay's underlay pointerDown felt potentially breaking since it assumes a lot about the existance of underlays and how the stacked underlays may be render w/ respect to each other (as siblings or as parent/children). Will re-evaluate when we come up with a more in depth way to determine if a external click should close only the top level overlay or the entire tree